py3status/modules/weather_owm.py

```diff
@@ -0,0 +1,739 @@
+            'mph': {
+                'speed': msec_speed * MPH_FROM_MSEC,
+                'gust': msec_gust * MPH_FROM_MSEC},
+            'kmh': {
+                'speed': msec_speed * KMH_FROM_MSEC,
+                'gust': msec_gust * KMH_FROM_MSEC}}
+
+        # Get the choice and add more
+        choice = options[self.wind_unit]
+        choice['icon'] = self.icons['wind']
+        choice['degree'] = wind['deg'] if ('deg' in wind) else 0
+
+        # Format the wind speed
+        return self.py3.safe_format(self.format_wind, choice)
+
+    def _format_humidity(self, wthr):
+        # Format the humidity (default zero humidity)
+        return self.py3.safe_format(self.format_humidity, {
+            'icon': self.icons['humidity'],
+            'humidity': self._jpath(wthr, OWM_HUMIDITY, 0),
+        })
+
+    def _format_pressure(self, wthr):
+        # Get data and add the icon
+        pressure = self._jpath(wthr, OWM_PRESSURE, dict())
+        pressure['icon'] = self.icons['pressure']
+
+        # Format the barometric pressure
+        return self.py3.safe_format(self.format_humidity, pressure)
+
+    def _format_temp(self, wthr):
+        # Get Kelvin data (default absolute zero)
+        kelvin = self._jpath(wthr, OWM_TEMP, 0)
+
+        # Temperature conversion methods
+        def kToC(val):
+            return val - 273.15
+
+        def kToF(val):
+            return val * (9.0 / 5.0) - 459.67
+
+        options = {
+            'c': {
+                'current': round(kToC(kelvin['temp'])),
+                'max': round(kToC(kelvin['temp_max'])),
+                'min': round(kToC(kelvin['temp_min']))},
+            'f': {
+                'current': round(kToF(kelvin['temp'])),
+                'max': round(kToF(kelvin['temp_max'])),
+                'min': round(kToF(kelvin['temp_min']))},
+            'k': {
+                'current': round(kelvin['temp']),
+                'max': round(kelvin['temp_max']),
+                'min': round(kelvin['temp_min'])}}
+
+        # Get the choice and add more
+        choice = options[self.temp_unit]
+        choice['icon'] = self.icons['temperature']
+        format_str = self.format_temperature
+
+        # Optionally add the color
+        color = None
+        if self.temp_color:
+            color = self.py3.threshold_get_color(choice['current'])
+            color_str = '\?color=%s' % color
+            format_str = color_str + ' ' + format_str
+
+        # Format the temperature
+        return self.py3.safe_format(format_str, choice)
+
+    def _format_sunrise(self, wthr):
+        # Get the time for sunrise (default is the start of time)
+        dt = datetime.datetime.utcfromtimestamp(
+            self._jpath(wthr, OWM_SUNRISE, 0))
+
+        # Format the sunrise
+        replaced = dt.strftime(self.format_sunrise)
+        return self.py3.safe_format(replaced, {
+            'icon': self.icons['sunrise'],
+        })
+
+    def _format_sunset(self, wthr):
```

Comment:
I'm going to change my IP API provider, there is another one which should give me the proper information! :)

Look for another commit!